t/t2806-config-cmd.t

```diff
@@ -193,5 +193,29 @@ test_expect_success 'flux-config reload fails as guest' '
 test_expect_success 'flux-config builtin works as guest' '
 	runas_guest flux config builtin rc1_path
 '
+test_expect_success 'flux-config get works when --config-path points to dir' '
+	mkdir -p altconfig &&
+	echo "flag = 99" >altconfig/foo.toml &&
+	flux config get --config-path=altconfig | jq -e .flag
+'
+test_expect_success 'flux-config get works when --config-path points to file' '
+	flux config get --config-path=altconfig/foo.toml | jq -e .flag
+'
+test_expect_success 'flux-config get fails when --config-path path is wrong' '
+	test_must_fail flux config get --config-path=/not/a/file
+'
+test_expect_success 'flux-config get fails when --config-path is invalid' '
+	echo "x x x" >badconf.toml &&
+	test_must_fail flux config get --config-path=badconf.toml
+'
+test_expect_success 'flux-config get --config-path=system fails' '
+	test_must_fail flux config get --config-path=system
+'
+test_expect_success 'flux-config get --config-path=security fails' '
+	test_must_fail flux config get --config-path=security
+'
+test_expect_success 'flux-config get --config-path=imp fails' '
+	test_must_fail flux config get --config-path=imp
+'
```

Comment:
Maybe a comment on why these tests should fail. I would have expected success?

Reply:
They are failing in CI so good point!

system/security/imp are aliases for compiled in paths so whether this succeeds or fails depends on whether good configs are installed at those paths.

I'm a little puzzled how to test.  Before your comments I pushed a change like this but now the test is kind of meaningless:
```
test_expect_success 'flux-config get --config-path=imp is special' '
        test_might_fail flux config get --config-path=imp 2>imp.err
        test -z imp.err || grep conf.d imp.err
'
```
Pondering.

Reply:
OK, I pushed that change.  It was slightly tricky to determine  the configured `$sysconfdir` except indirectly through  `flux config builtin`, but `flux config builtin` will always give in-tree paths when run from the build tree.

I added `--intree` and `--installed` options to `flux config builtin` and then take the dirname of `flux config builtin --installed rc1_path`.  Not super satisfying :shrug:

Reply:
well that seems useful to me for testing, and now the tests actually test something, so seems good to me!